### ISSUE TITLE
Bring in the JUnit Assume feature to Test Case `testSubscription`

### ISSUE THREAD

**Issue by Codegass** [apache/dubbo#10213]
- [X] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.

## Describe the proposal

Hi Dubbo community,

I have some suggestions about bringing in the Assume feature to the project. In JUnit 4&5, the framework brings the Assume functions to replace Assertion in precondition checking. As they mentioned in the document (you can check it [here](https://junit.org/junit5/docs/5.0.0/api/org/junit/jupiter/api/Assumptions.html)):

> Assume functions is a set of methods useful for stating assumptions about the conditions in which a test is meaningful. A failed assumption does not mean the code is broken, but that the test provides no useful information.

By bringing in this Assume feature, I believe it can help to improve the test case [testSubscription](https://github.com/apache/dubbo/blob/d41a0448c70ac11e1d3df4b80ba9fd00829097a6/dubbo-registry/dubbo-registry-multiple/src/test/java/org/apache/dubbo/registry/multiple/MultipleRegistry2S2RTest.java#L140).

Right now this case is asserting the precondition([`providerList`](https://github.com/apache/dubbo/blob/d41a0448c70ac11e1d3df4b80ba9fd00829097a6/dubbo-registry/dubbo-registry-multiple/src/test/java/org/apache/dubbo/registry/multiple/MultipleRegistry2S2RTest.java#L147-L148)) before the actual testing target(`multipleRegistry.subscribe()`) is executed.

So when the test fails, it may due to 2 reasons:

* The functions related to the testing precondition (here is `providerList`) have bugs.
* The functions related to `multipleRegistry.subscribe()` have bugs. (What we actually want to check with this test)

As a test case, it should focus on the test target itself, not the preconditions. The preconditions should be checked by other tests and should not let the test fails. So here I suggest replacing the Assert function and `System.out.println` in [line 147-148](https://github.com/apache/dubbo/blob/d41a0448c70ac11e1d3df4b80ba9fd00829097a6/dubbo-registry/dubbo-registry-multiple/src/test/java/org/apache/dubbo/registry/multiple/MultipleRegistry2S2RTest.java#L147-L148) with Assume functions.

Before
```java
        String path = "/dubbo/" + SERVICE_NAME + "/providers";
        List<String> providerList = zookeeperClient.getChildren(path);
        Assertions.assertTrue(!providerList.isEmpty());
        System.out.println(providerList.get(0));
```

After
```java
        String path = "/dubbo/" + SERVICE_NAME + "/providers";
        List<String> providerList = zookeeperClient.getChildren(path);
        Assumptions.assumingThat(providerList.get(0), is(`http2%3A%2F%2Fmultiple%2Forg.apache.dubbo.registry.MultipleService2S2R2%3Fcategory%3Dproviders%26methods%3Dtest1%2Ctest2%26notify%3Dfalse`));
```

After the replacement, when running this test case, developers can focus on the testing targets, and clearly identify the testing failure source.

Please let me know if you think this proposal makes sense. If so, then I can try to submit a PR accordingly. If not, comments are appropriate.

**Comment by AlbumenJ** [comment]
Good idea. Would you please a PR?

**Comment by Codegass** [comment]
Thanks! I will try to submit the PR today. It is encouraging to see my proposal is agreed.

**Comment by wangchengming666** [comment]
go ahead~

**Comment by wangchengming666** [comment]
fixed by #10413
